src/covers/discogscoverprovider.cpp

```diff
@@ -0,0 +1,373 @@
+/* This file is part of Clementine.
+   Copyright 2012, Martin Björklund <mbj4668@gmail.com>
+   Copyright 2018, Jonas Kvinge <jonas@jkvinge.net>
+
+   Clementine is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   Clementine is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Clementine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <QByteArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QList>
+#include <QNetworkReply>
+#include <QNetworkRequest>
+#include <QPair>
+#include <QString>
+#include <QStringList>
+#include <QUrlQuery>
+#include <QVariant>
+
+#include "discogscoverprovider.h"
+
+#include "core/closure.h"
+#include "core/logging.h"
+#include "core/network.h"
+#include "core/utilities.h"
+
+const char* DiscogsCoverProvider::kUrlSearch =
+    "https://api.discogs.com/database/search";
+const char* DiscogsCoverProvider::kUrlReleases =
+    "https://api.discogs.com/releases";
+
+const char* DiscogsCoverProvider::kAccessKeyB64 =
+    "YVR4Yk5JTnlmUkhFY0pTaldid2c=";
+const char* DiscogsCoverProvider::kSecretKeyB64 =
+    "QkJNb2tMVXVUVFhSRWRUVmZDc0ZGamZmSWRjdHZRVno=";
+
+DiscogsCoverProvider::DiscogsCoverProvider(QObject* parent)
+    : CoverProvider("Discogs", false, parent),
+      network_(new NetworkAccessManager(this)) {}
+
+bool DiscogsCoverProvider::StartSearch(const QString& artist,
+                                       const QString& album, int s_id) {
+  DiscogsCoverSearchContext* s_ctx = new DiscogsCoverSearchContext;
+  if (s_ctx == nullptr) return false;
```

Comment:
This can never be `nullptr` so there's no point in the check

Reply:
Agree